metrics.go

```diff
@@ -0,0 +1,138 @@
+/*
+ Copyright © 2023 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package pmax
+
+import (
+	"context"
+	"encoding/json"
+	types "github.com/dell/gopowermax/v2/types/v100"
+	"net/http"
+	"time"
+)
+
+// The following constants are for the query of performance metrics for pmax
+const (
+	Average      = "Average"
+	Performance  = "performance"
+	StorageGroup = "/StorageGroup"
+	Volume       = "/Volume"
+	Metrics      = "/metrics"
+	Keys         = "/keys"
+	Array        = "/Array"
+)
+
+// GetStorageGroupPerfKeys returns the available timestamp for the storage group performance
+func (c *Client) GetStorageGroupPerfKeys(ctx context.Context, symID string) (*types.StorageGroupKeysResult, error) {
+	defer c.TimeSpent("GetStorageGroupPerfKeys", time.Now())
+	if _, err := c.IsAllowedArray(symID); err != nil {
+		return nil, err
+	}
+	URL := RESTPrefix + Performance + StorageGroup + Keys
+	ctx, cancel := c.GetTimeoutContext(ctx)
+	defer cancel()
+	params := types.StorageGroupKeysParam{
+		SymmetrixID: symID,
+	}
+	resp, err := c.api.DoAndGetResponseBody(ctx, http.MethodPost, URL, c.getDefaultHeaders(), params)
+	defer resp.Body.Close()
+	if err = c.checkResponse(resp); err != nil {
+		return nil, err
+	}
+	storageGroupInfo := &types.StorageGroupKeysResult{}
+	decoder := json.NewDecoder(resp.Body)
+	if err = decoder.Decode(storageGroupInfo); err != nil {
+		return nil, err
+	}
+	return storageGroupInfo, nil
+}
+
+// GetArrayPerfKeys returns the available timestamp for the array performance
+func (c *Client) GetArrayPerfKeys(ctx context.Context) (*types.ArrayKeysResult, error) {
```

Comment:
@P-Cao , is there an API returning current timestamp of PowerMax?

Reply:
> @P-Cao , is there an API returning current timestamp of PowerMax?

Didn't find one under system scope

Reply:
OK, we should mention that our pod and PowerMax should use the same time server in document.